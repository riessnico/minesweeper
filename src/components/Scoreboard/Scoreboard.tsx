import React from 'react';
import styled from '@emotion/styled';

import { Counter } from './Counter';
import { Level } from './Level';
import { Reset } from './Reset';

export interface ScoreboardProps {
  /**
   * Timer
   */
  time: string;
  /**
   * possible game scenarios
   */
  levels: string[];
  /**
   * action handler when the gamereset button is clicked
   */
  onReset: () => void;
  /**
   * Bombs in the field
   */
  mines: string;
}

export const Scoreboard: React.FC<ScoreboardProps> = ({
  levels,
  mines,
  onReset,
  time,
}) => (
  <Wrapper>
    <Counter>{time}</Counter>
    <Level>{levels}</Level>
    <Reset onReset={onReset} />
    <Counter>{mines}</Counter>
  </Wrapper>
);

const Wrapper = styled.div`
  display: flex;
  width: max-content;
  padding-bottom: 2vw;
  justify-content: space-between;
`;
