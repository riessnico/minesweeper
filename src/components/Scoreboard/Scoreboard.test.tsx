import React from 'react';
import { render } from '@testing-library/react';

import { Scoreboard } from './Scoreboard';

it('Scoreboard renders correctly', () => {
  const { asFragment } = render(
    <Scoreboard
      time="000"
      levels={['beginner', 'intermediate', 'expert']}
      mines="010"
      onReset={() => null}
    />,
  );

  expect(asFragment()).toMatchSnapshot();
});
