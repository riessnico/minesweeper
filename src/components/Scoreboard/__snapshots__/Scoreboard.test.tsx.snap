// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1gexv9m"
  >
    <div
      class="css-d2q1pa"
    >
      000
    </div>
    <select
      class="css-160ys9i"
    >
      <option
        class="css-zcmdxr"
        value="beginner"
      >
        beginner
      </option>
      <option
        class="css-zcmdxr"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-zcmdxr"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-mt04f9"
    >
      ðŸ˜Š
    </button>
    <div
      class="css-d2q1pa"
    >
      010
    </div>
  </div>
</DocumentFragment>
`;
